go-backend/apperrors/apperrors.go

```diff
@@ -0,0 +1,106 @@
+/*
+========================================================================
+Error Definitions
+	This file, part of the "apperrors" package, defines all the error
+	variables by name that'll be used throughout the entire application.
+
+Rules
+	Every error variable MUST start with "Err" - with a capital E so
+	we can export it for use in other packages.
+
+	Additionally, please place a comment (one or more lines) on the line
+	before the declaration of the error variable that explains what that
+	variable is intended to be used for, or the error concept it's meant
+	to represent.
+
+Notes
+	There are also additional helper functions in this file pertaining to
+	miscellaneous error handling.
+========================================================================
+*/
+
+package apperrors
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+// ErrorStruct - a generic struct you can use to create error messages/logs to be converted
+// to JSON or other types of messages/data as you need it
+type ErrorStruct struct {
+	Message string `json:"message,omitempty"` // Your message to the end user or developer
+	Status  int    `json:"status,omitempty"`  // HTTP status code that should go with the message/log (if any)
+}
+
+// JSONError - This function writes out an error response with the status
+// header passed in
+func JSONError(rw http.ResponseWriter, status int, err error) {
+	// Create the ErrorStruct object for later use
+	errObj := ErrorStruct{
+		Message: err.Error(),
+		Status:  status,
+	}
+
+	errJSON, _ := json.Marshal(&errObj)
```

Comment:
I know there are very less chances that this call will return an error, but we should ignore this error here.

Reply:
Fixed; it's now checking for an error and warning if it's encountered.